text/src/autogluon/text/automm/optimization/utils.py

```diff
@@ -211,11 +211,34 @@ def get_weight_decay_param_names(model: nn.Module):
     -------
     A list of parameter names not using weight decay.
     """
-    decay_param_names = get_parameter_names(model, [nn.LayerNorm])
+    # By default, we should not apply weight decay for all the norm layers
+    decay_param_names = get_parameter_names(model,
+                                            [nn.LayerNorm, nn.BatchNorm1d, nn.BatchNorm2d, nn.BatchNorm3d,
+                                             nn.GroupNorm])
     decay_param_names = [name for name in decay_param_names if "bias" not in name]
     return decay_param_names
 
 
+def get_norm_layer_param_names(model: nn.Module):
+    """Get parameters associated with the normalization layers
+
+    Parameters
+    ----------
+    model
+        A Pytorch model
+
+    Returns
+    -------
+    norm_param_names
+        A list of names that
+    """
+    all_param_names = get_parameter_names(model, [])
```

Comment:
`all_param_names = [name for name, _ in model.named_parameters()]` seems more straightforward.